js/src/components/TopContentNavigation.js

```diff
@@ -0,0 +1,233 @@
+import { coursePropTypes } from '@anu/utilities/transform.course';
+import Typography from '@material-ui/core/Typography';
+import LinearProgress from '@material-ui/core/LinearProgress';
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    background: theme.palette.grey[200],
+    boxSizing: 'border-box',
+    padding: theme.spacing(0, 0.25, 0, 4),
+    marginLeft: theme.spacing(0.5),
+    marginBottom: theme.spacing(8),
+    marginTop: 0,
+    [theme.breakpoints.down('sm')]: {
+      marginLeft: -theme.spacing(1),
+      marginRight: -theme.spacing(1),
+      marginTop: -theme.spacing(1),
+      paddingRight: 0,
+    },
+  },
+  stickyContainer: {
+    position: 'fixed',
+    top: 0,
+    zIndex: 10,
+    marginTop: theme.spacing(0),
+    [theme.breakpoints.down('sm')]: {
+      width: `100%!important`,
+    },
+  },
+  emptyContainer: {
+    height: '115px',
+  },
+  actionsSection: {
+    display: 'flex',
+    justifyContent: 'end',
+    marginBottom: theme.spacing(1.5),
+    marginTop: theme.spacing(1.5),
+  },
+  titleSection: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+  titleWrapper: {
+    paddingRight: '250px',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    [theme.breakpoints.down('lg')]: {
+      maxWidth: '700px',
+    },
+    [theme.breakpoints.down('md')]: {
+      maxWidth: '400px',
+    },
+  },
+  navigationMobileSection: {
+    padding: '0!important',
+  },
+  pageNumberSection: {
+    display: 'flex',
+    alignItems: 'center',
+    marginRight: theme.spacing(2),
+  },
+  progressSection: {
+    marginLeft: -theme.spacing(4),
+    marginRight: -theme.spacing(0.25),
+    maxWidth: `calc(100% + ${theme.spacing(4)}px)`,
+    flexBasis: `calc(100% + ${theme.spacing(4)}px)`,
+  },
+}));
+
+// Sticky top navigation used for lessons and quizzes.
+const ContentTopNavigation = ({
+  isIntro,
+  sections,
+  currentLesson,
+  nextLesson,
+  prevLesson,
+  currentIndex,
+  isEnabled,
+  course,
+  stepsDirection,
+}) => {
+  const classes = useStyles();
+
+  const [isSticky, _setIsSticky] = useState(false);
+
+  // Hooks to track the "isSticky" value. Note that we have to use useRef here as
+  // well as useState, in order to have the step data accessible in our
+  // event listeners which normally don't get updates from change of the state.
+  // See https://file-translate.com/en/blog/react-state-in-event.
+  const isStickyRef = useRef(isSticky);
+
+  const setIsSticky = (value) => {
+    isStickyRef.current = value;
+    _setIsSticky(value);
+  };
+
+  useEffect(() => {
+    const onScroll = () => {
+      // "emptyNavbar" is empty block with predefined height, used to avoid
+      // visual collision during moving between "static" and "sticky" states
+      // for top navigation.
+      const stickyNavbar = document.getElementById('top-content-navigation');
+      const emptyNavbar = document.getElementById('top-empty-navigation');
```

Comment:
Why can't we increase padding of the container instead of extra element on the page?